austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,155 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/////////////////////////  INITIAL  STUFF  /////////////////////////
+/mob/living/simple_animal/hostile/mining_drone
+	var/beacons = 15  //maximum beacons that any one drone can drop
+	var/emagged = FALSE  //lets minebot shoot sentient life
```

Comment:
You don't really need a var for this, you can just check the object flags

i.e `if(obj_flags & EMAGGED)`

Reply:
mobs don't have an object flags variable.